composinator/convert.go

```diff
@@ -0,0 +1,517 @@
+	}
+	stack, err := client.Stack.ById(stackID)
+	if err != nil {
+		return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get stack")
+	}
+	servicesToExport := []*v3.Service{}
+	// lbcertMap is used for certificate reference by lbConfig
+	lbcertMap := map[string]v3.Certificate{}
+	// serviceMap is used for reference by portRule
+	serviceMap := map[string]v3.Service{}
+	containerMap := map[string]v3.Instance{}
+	secretMap := map[string]v3.Secret{}
+	for _, id := range stack.ServiceIds {
+		service, err := client.Service.ById(id)
+		if err != nil {
+			return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get service")
+		}
+		if service.LbConfig != nil {
+			// prepare certMap
+			if service.LbConfig.DefaultCertificateId != "" {
+				cert, err := client.Certificate.ById(service.LbConfig.DefaultCertificateId)
+				if err != nil {
+					return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get lbconfig")
+				}
+				lbcertMap[cert.Id] = *cert
+			}
+			if len(service.LbConfig.CertificateIds) > 0 {
+				for _, certID := range service.LbConfig.CertificateIds {
+					cert, err := client.Certificate.ById(certID)
+					if err != nil {
+						return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get lbconfig")
+					}
+					lbcertMap[cert.Id] = *cert
+				}
+			}
+			// prepare serviceMap and containerMap
+			for _, portRule := range service.LbConfig.PortRules {
+				if portRule.ServiceId != "" {
+					service, err := client.Service.ById(portRule.ServiceId)
+					if err != nil {
+						return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get service")
+					}
+					serviceMap[service.Id] = *service
+				}
+				if portRule.InstanceId != "" {
+					instance, err := client.Instance.ById(portRule.InstanceId)
+					if err != nil {
+						return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get instance")
+					}
+					containerMap[instance.Id] = *instance
+				}
+			}
+		}
+		// prepare secretMap
+		for _, launchConfig := range append([]v3.LaunchConfig{*service.LaunchConfig}, service.SecondaryLaunchConfigs...) {
+			for _, sr := range launchConfig.Secrets {
+				secret, err := client.Secret.ById(sr.SecretId)
+				if err != nil {
+					return nil, nil, nil, nil, nil, nil, errors.Wrap(err, "can't get secret")
+				}
+				secretMap[secret.Id] = *secret
+			}
+		}
+
+		servicesToExport = append(servicesToExport, service)
+	}
+	volumeTemplatesMap := map[string]v3.VolumeTemplate{}
+	for _, vt := range volumeTemplates.Data {
+		volumeTemplatesMap[vt.Name] = vt
+	}
+	return servicesToExport, volumeTemplatesMap, lbcertMap, serviceMap, containerMap, secretMap, nil
+}
+
+func createComposeData(services []*v3.Service, volumeTemplates map[string]v3.VolumeTemplate, certMap map[string]v3.Certificate,
+	serviceMap map[string]v3.Service, containerMap map[string]v3.Instance, secretMap map[string]v3.Secret, format string) (string, string, string, error) {
+	if format == "combined" {
+		compose := config.NewConfig()
```

Comment:
since this method is just one huge if statement, extract the bodies of the if into methods.  so you just have something closer to

```golang
if format == "combined" {
  createCombinedComposeData()
} else {
  createSplitComposeData()
}
```